pkg/txn/client/client.go

```diff
@@ -313,20 +320,50 @@ func (client *txnClient) popTransaction(txn txn.TxnMeta) {
 	client.removeFromLeakCheck(txn.ID)
 }
 
-func (client *txnClient) pushTransaction(txn txn.TxnMeta) {
+func (client *txnClient) pushTransaction(txn txn.TxnMeta) error {
 	client.mu.Lock()
 	defer client.mu.Unlock()
-	i := sort.Search(len(client.mu.txns), func(i int) bool {
-		return client.mu.txns[i].SnapshotTS.GreaterEq(txn.SnapshotTS)
-	})
-	if i == len(client.mu.txns) {
-		client.mu.txns = append(client.mu.txns, txn)
-	} else {
-		client.mu.txns = append(client.mu.txns[:i+1], client.mu.txns[i:]...)
-		client.mu.txns[i] = txn
+
+	if client.mu.cnReady {
+		i := sort.Search(len(client.mu.txns), func(i int) bool {
+			return client.mu.txns[i].SnapshotTS.GreaterEq(txn.SnapshotTS)
+		})
+		if i == len(client.mu.txns) {
+			client.mu.txns = append(client.mu.txns, txn)
+		} else {
+			client.mu.txns = append(client.mu.txns[:i+1], client.mu.txns[i:]...)
+			client.mu.txns[i] = txn
+		}
+		if client.mu.minTS.IsEmpty() || txn.SnapshotTS.Less(client.mu.minTS) {
+			client.mu.minTS = txn.SnapshotTS
+		}
+		return nil
 	}
-	if client.mu.minTS.IsEmpty() || txn.SnapshotTS.Less(client.mu.minTS) {
-		client.mu.minTS = txn.SnapshotTS
+	return moerr.NewInternalErrorNoCtx("cn service is not ready, plz retry later")
+}
+
+func (client *txnClient) Pause() {
+	client.mu.Lock()
+	defer client.mu.Unlock()
+
+	logutil.Infof("cn ready status changed, ready: false")
+	client.mu.cnReady = false
```

Comment:
rename cnReady to state. Set state to false if pause called

Reply:
ok.